apps/dashboard/app/lib/smart_attributes/attributes/bc_num_hours.rb

```diff
@@ -10,10 +10,10 @@ def self.build_bc_num_hours(opts = {})
 
   module Attributes
     class BcNumHours < Attribute
-      # Hash of options used to define this attribute
-      # @return [Hash] attribute options
-      def opts
-        @opts.reverse_merge(min: 1, step: 1)
+
+      def initialize(id, opts)
+        super(id, opts)
+        @opts = @opts.reverse_merge(min: 1, step: 1)
```

Comment:
While this PR is ready for review - I'm not going to merge it until we tie up a few loose ends with and release `v3.0.1`.

Specifically because of this and the edit to the parent class as well. I'm fairly confident it still works as expected, but I don't want some bug leaking into release 3.0.

So, once 3.0.1 is sorted and it's on it's on release branch, we can pull this into the mainline which will by then effectively be the start of 3.1.